soh/src/overlays/actors/ovl_player_actor/z_player.c

```diff
@@ -12715,30 +12738,69 @@ s32 func_8084DFF4(GlobalContext* globalCtx, Player* this) {
             gSaveContext.swordHealth = 8;       
         }
 
-        Message_StartTextbox(globalCtx, giEntry->textId, &this->actor);
-        Item_Give(globalCtx, giEntry->itemId);
-        
-        Player_SetPendingFlag(this, globalCtx);
-
-        if (((this->getItemId >= GI_RUPEE_GREEN) && (this->getItemId <= GI_RUPEE_RED)) ||
-            ((this->getItemId >= GI_RUPEE_PURPLE) && (this->getItemId <= GI_RUPEE_GOLD)) ||
-            ((this->getItemId >= GI_RUPEE_GREEN_LOSE) && (this->getItemId <= GI_RUPEE_PURPLE_LOSE)) ||
-            (this->getItemId == GI_HEART)) {
-            Audio_PlaySoundGeneral(NA_SE_SY_GET_BOXITEM, &D_801333D4, 4, &D_801333E0, &D_801333E0, &D_801333E8);
+        Message_StartTextbox(globalCtx, giEntry.textId, &this->actor);
+        if (giEntry.modIndex == 0) {
+            Item_Give(globalCtx, giEntry.itemId);
         }
         else {
-            if ((this->getItemId == GI_HEART_CONTAINER_2) || (this->getItemId == GI_HEART_CONTAINER) ||
-                ((this->getItemId == GI_HEART_PIECE) &&
-                    ((gSaveContext.inventory.questItems & 0xF0000000) == 0x40000000))) {
-                temp1 = NA_BGM_HEART_GET | 0x900;
+            Randomizer_Item_Give(globalCtx, giEntry);
+        }
+        Player_SetPendingFlag(this, globalCtx);
+
+        if (this->getItemEntry.objectId != OBJECT_INVALID) {
+            if (giEntry.modIndex == 0) {
+                if (((this->getItemId >= GI_RUPEE_GREEN) && (this->getItemId <= GI_RUPEE_RED)) ||
+                    ((this->getItemId >= GI_RUPEE_PURPLE) && (this->getItemId <= GI_RUPEE_GOLD)) ||
+                    ((this->getItemId >= GI_RUPEE_GREEN_LOSE) && (this->getItemId <= GI_RUPEE_PURPLE_LOSE)) ||
+                    (this->getItemId == GI_HEART)) {
+                    Audio_PlaySoundGeneral(NA_SE_SY_GET_BOXITEM, &D_801333D4, 4, &D_801333E0, &D_801333E0, &D_801333E8);
+                } else {
+                    if ((this->getItemId == GI_HEART_CONTAINER_2) || (this->getItemId == GI_HEART_CONTAINER) ||
+                        ((this->getItemId == GI_HEART_PIECE) &&
+                         ((gSaveContext.inventory.questItems & 0xF0000000) == 0x40000000))) {
+                        temp1 = NA_BGM_HEART_GET | 0x900;
+                    } else {
+                        temp1 = temp2 =
+                            (this->getItemId == GI_HEART_PIECE) ? NA_BGM_SMALL_ITEM_GET : NA_BGM_ITEM_GET | 0x900;
+                    }
+                    Audio_PlayFanfare(temp1);
+                }
+            } else {
```

Comment:
should probably else if here

Reply:
Addressed by 0c334615b65abe06a7d7e9a3ceb373c6e28e7203